server/src/main/java/org/candlepin/model/Product.java

```diff
@@ -104,10 +104,16 @@
     @LazyCollection(LazyCollectionOption.EXTRA) // allows .size() without loading all data
     private List<ProductContent> productContent;
 
+    /*
+     * hibernate persists empty set as null, and tries to fetch
+     * dependentProductIds upon a fetch when we lazy load. to fix this, we eager
+     * fetch.
+     */
     @ElementCollection
     @CollectionTable(name = "cpo_product_dependent_products",
                      joinColumns = @JoinColumn(name = "product_uuid"))
     @Column(name = "element")
+    @LazyCollection(LazyCollectionOption.FALSE)
```

Comment:
That link looks to be referring to storing nulls at the end of the list and not getting them back, and getting an empty list out when it only contained null. I think we're hitting a different problem entirely.